File: day01/python/solution.py
cals = [[]]
with open("input.txt") as f:
    [
        cals[-1].append(int(x.strip())) if x != "\n" else cals.append([])
        for x in f.readlines()
    ]
    cals = sorted(map(sum, cals))

print(f"Part 1: {sorted(cals)[-1]}")  # 70116
print(f"Part 2: {sum(sorted(cals)[-3:])}")  # 206582

File: day01/clojure/src/solution.clj
(ns solution
  (:require [clojure.string :as str]))

(defn common []
  (->> (slurp "input.txt")
       (#(str/split % #"\n\n"))
       (mapv #(str/split % #"\n"))
       (mapv #(mapv parse-long %))
       (mapv #(reduce + %))))

(defn part1-sol []
  (->> (common) (sort >) (first)))

(defn part2-sol []
  (->> (common) (sort >) (take 3) (reduce +)))

(defn -main
  "Invoke me with clojure -M -m solution"
  [& _]
  (println (str "Part 1: " (part1-sol)))
  (println (str "Part 2: " (part2-sol))))
